core/unix/memquery.c

```diff
@@ -144,6 +145,9 @@ memquery_library_bounds_by_iterator(const char *name, app_pc *start /*IN/OUT*/,
                         /* we keep the last '/' at end */
                         ++slash;
                         strncpy(dst, src, MIN(dstsz, (slash - src)));
+                        if (filename != NULL)
+                            /* slash is filename */
+                            strncpy(filename, slash, MIN(strlen(slash), filename_size));
```

Comment:
bug: fails to set terminating null char if it maxes out (done below for dst)

Reply:
Done